overlord/snapstate/snapstate.go

```diff
@@ -410,21 +410,38 @@ func Update(s *state.State, name, channel string, userID int, flags Flags) (*sta
 		channel = snapst.Channel
 	}
 
-	updateInfo, err := updateInfo(s, &snapst, channel, userID, flags)
-	if err != nil {
-		return nil, err
+	ss := &SnapSetup{
+		Channel: channel,
+		UserID:  userID,
+		Flags:   SnapSetupFlags(flags),
 	}
-	if err := checkRevisionIsNew(name, &snapst, updateInfo.Revision); err != nil {
```

Comment:
note that this removal means ABA refreshes are unblocked but unhandled (--> bad things could happen)